src/endpoint/s3/ops/s3_post_bucket_delete.js

```diff
@@ -29,7 +30,7 @@ async function post_bucket_delete(req) {
     const uniq_map = new Map();
     for (const item of delete_list) {
         const key = item.Key?.[0];
-        const version_id = item.VersionId?.[0];
+        const version_id = s3_utils.parse_version_id(item.VersionId?.[0]);
```

Comment:
did you check that AWS also errors in this api when sent with `<VersionId></VersionId>`?

Reply:
I checked this one and this should be reverted, see other comment for details.